units/provisioning_agent_deploy.go

```diff
@@ -63,7 +63,7 @@ func NewAgentDeployJob(env evergreen.Environment, h host.Host, id string) amboy.
 	j.HostID = h.Id
 	j.env = env
 	j.SetPriority(1)
-	j.SetID(fmt.Sprintf("%s.%s.%s", agentDeployJobName, j.HostID, id))
+	j.SetID(fmt.Sprintf("%s.%s.attempt-%d.%s", agentDeployJobName, j.HostID, h.AgentDeployAttempt, id))
```

Comment:
Agent deploy attempt is already in the id.

Reply:
removed the instance in crons, so I think this one should stay.